R/drive_mkdir.R

```diff
@@ -9,13 +9,27 @@
 drive_mkdir <- function(dir = NULL, path = NULL, verbose = TRUE) {
   parent <- NULL
 
-  ## TO DO: accept path input in the generalized "Drive file" style
-  ## and check that it's a folder
   if (!is.null(path)) {
-    path <- append_slash(path)
-    parent <- get_one_path(path)$id
+    if (inherits(path, "dribble") || inherits(path, "drive_id")) {
+      path <- as_dribble(path)
+      if (!is_folder(path)){
+        stop(
+          glue::glue_data(path, "'path' is not a folder:\n{name}"),
+          call. = FALSE
+        )
+      }
+    } else {
+      path <- append_slash(path)
+      path <- get_one_path(path)
+      if (path$mimeType != "application/vnd.google-apps.folder") {
```

Comment:
Yeah I think it should work just like the parent folder in `drive_mv()`. If it's not simple, then it means we need to improve something else. I need to refactor the path stuff in light of all the other changes. Will do shortly.